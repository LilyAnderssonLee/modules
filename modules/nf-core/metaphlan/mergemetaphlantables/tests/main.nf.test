nextflow_process {

    name "Test Process METAPHLAN_MERGEMETAPHLANTABLES"
    script "../main.nf"
    process "METAPHLAN_MERGEMETAPHLANTABLES"

    tag "modules"
    tag "modules_nfcore"
    tag "untar"
    tag "metaphlan"
    tag "metaphlan/metaphlan"
    tag "metaphlan/mergemetaphlantables"


    test("sarscov2 - illumina single end [fastq]") {
        setup {
            run("UNTAR") {
                script "../../../untar/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [],
                        file( params.modules_testdata_base_path + 'delete_me/metaphlan4_database.tar.gz', checkIfExists: true )
                    ])
                    """
                }
            }
            run("METAPHLAN_METAPHLAN") {
                script "../../../metaphlan/metaphlan/main.nf"
                process {
                    """
                    input[0] = Channel.of(
                        [ [ id: 'test', single_end: true],
                        file( params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz',checkIfExists: true ) ]
                    )
                    input[1] = UNTAR.out.untar.map{ it[1] }
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = METAPHLAN_MERGEMETAPHLANTABLES.out.profile
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot( process.out ).match() }
            )
        }
    }


    test("sarscov2 - illumina single end [fastq] - stub") {

        options "-stub"

        setup {
            run("UNTAR") {
                script "../../../untar/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [],
                        file( params.modules_testdata_base_path + 'delete_me/metaphlan4_database.tar.gz', checkIfExists: true )
                    ])
                    """
                }
            }
            run("METAPHLAN_METAPHLAN") {
                script "../../../metaphlan/metaphlan/main.nf"
                process {
                    """
                    input[0] = Channel.of(
                        [[ id: 'test', single_end: true],
                        file( params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz',checkIfExists: true ) ]
                    )
                    input[1] = UNTAR.out.untar.map{ it[1] }
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = METAPHLAN_METAPHLAN.out.profile
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot( process.out ).match() }
            )
        }
    }
}
