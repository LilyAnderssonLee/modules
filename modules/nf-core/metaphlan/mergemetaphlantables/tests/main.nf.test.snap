{
    "sarscov2 - illumina single end [fastq] - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "txt": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-13T14:09:23.436275"
    }
}